### PR TITLE
add noDev option. closes #5

### LINKED ISSUE #5: ignore devDependencies by an option
see https://github.com/nodesource/ncm-cli/issues/69

This would really help the usefulness of ncm-cli for assessing what you are actually distributing (e.g. ncm-cli itself!).

### PR DESCRIPTION
**Pull request by juliangruber**

### CODE REVIEW

File: example.js [hunk 1]
```diff
@@ -2,10 +2,13 @@
...
 const stringify = require('json-stringify-safe')
+const minimist = require('minimist')
 
 const main = async () => {
-  const dir = process.argv[2] || __dirname
-  const tree = await getTree(dir)
+  const argv = minimist(process.argv.slice(2), { boolean: 'dev' })
+  const dir = argv._[0] || __dirname
+  const { dev } = argv
+  const tree = await getTree(dir, { dev })
```
Comment (Raynos) [inline example.js]: nitpick: `const tree = await getTree(dir, { dev: argv.dev })` as a one liner.

File: index.js [hunk 2]
```diff
@@ -124,7 +126,7 @@ const getTreeFromYarnLock = ({ yarnLock: yarnLockString, packageJSON }) => {
   return tree
 }
 
-const getTreeFromNodeModules = async dir => {
+const getTreeFromNodeModules = async (dir, { dev = true } = {}) => {
```
Comment (Raynos) [inline index.js]: nitpick: avoid optional options if possible.

File: README.md [hunk 3]
```diff
@@ -63,13 +63,17 @@ $ npm install universal-module-tree
 
 ## API
 
-### getTree(dir) => `Promise`
-### getTree.fromPackageLock({ packageLock, packageJSON }) => `Object`
-### getTree.fromYarnLock({ yarnLock, packageJSON }) => `Object`
-### getTree.fromNodeModules(path) => `Promise`
+### getTree(dir, { dev = true }) => `Promise`
```
Comment (Raynos) [inline README.md]: booleans that default to true should really be renamed and default to false aka `skipDev = false`

**Review by Raynos — APPROVED** [review]

File: index.js (hunk 2 above)
Comment (Raynos) [reply index.js]: i think its fine for `dev = true` to be optional; but ideally all callers to `getTreeFromNodeModules` pass in `{ dev: bool }` as a required argument.

Too many optional parameters leads to accidentally not passing opts.dev down to the next function call.

**Review by Raynos — APPROVED** [review]

**Comment by juliangruber** [comment]
3.1.0

**Pull request merged**